llvm/include/llvm/Support/BlockFrequency.h

```diff
@@ -76,6 +78,12 @@ class BlockFrequency {
     return NewFreq;
   }
 
+  /// Multiplies frequency with `Factor` and stores the result into `Result`.
+  /// Returns `true` if an overflow occured. Overflows are common and should be
+  /// checked by all callers.
+  bool mul(uint64_t Factor,
```

Comment:
Could we use `[[nodiscard]] bool` instead here?  We shouldn't have to use `__attribute__((warn_unused_result))` as `[[nodiscard]]` is available as part of C++17.

Reply:
Somehow I know `__attribute__((...))`s better than C++17 :) Will switch, great to see this feature in the standard!